lib/embedded_ansible.rb

```diff
@@ -1,53 +1,27 @@
-require "securerandom"
-require "awesome_spawn"
-require "linux_admin"
-require "ansible_tower_client"
-require "fileutils"
-
 class EmbeddedAnsible
+  include Vmdb::Logging
+
   ANSIBLE_ROLE           = "embedded_ansible".freeze
-  SETUP_SCRIPT           = "ansible-tower-setup".freeze
-  SECRET_KEY_FILE        = "/etc/tower/SECRET_KEY".freeze
-  SETTINGS_FILE          = "/etc/tower/settings.py".freeze
-  EXCLUDE_TAGS           = "packages,migrations,firewall".freeze
-  HTTP_PORT              = 54_321
-  HTTPS_PORT             = 54_322
   WAIT_FOR_ANSIBLE_SLEEP = 1.second
-  TOWER_VERSION_FILE     = "/var/lib/awx/.tower_version".freeze
-  ANSIBLE_DC_NAME        = "ansible".freeze
-
-  def self.available?
-    return true if MiqEnvironment::Command.is_container?
-    return false unless MiqEnvironment::Command.is_appliance?
 
-    required_rpms = Set["ansible-tower-server", "ansible-tower-setup"]
-    required_rpms.subset?(LinuxAdmin::Rpm.list_installed.keys.to_set)
+  def self.new
+    require "ansible_tower_client"
+    self == EmbeddedAnsible ? detect_available_platform.new : super
   end
 
-  def self.enabled?
-    MiqServer.my_server(true).has_active_role?(ANSIBLE_ROLE)
+  def self.detect_available_platform
+    subclasses.detect(&:available?) || NullEmbeddedAnsible
```

Comment:
I don't believe there is any sort order to the subclasses, so if there are multiple available, which one will it pick?  Is it random?  I'm wondering if we should add in a priority method like we do with the PDF plugins.